tokio-epoll-uring/src/system/tests.rs

```diff
@@ -88,3 +88,103 @@ async fn basic() {
 
     system.initiate_shutdown().await;
 }
+
+// This test changes & observes process-wide state.
+// To avoid requiring cargo nextest / --test-threads 1, we do some trickery.
+// TODO: find means to avoid this trickery / make it more robust.
+#[tokio::test]
+async fn hitting_memlock_limit_does_not_panic() {
+    let max_number_of_systems_spawned_by_other_tests: usize = 100; // other tests affect VmLck as well.
+
+    let (soft, hard) =
+        nix::sys::resource::getrlimit(nix::sys::resource::Resource::RLIMIT_MEMLOCK).unwrap();
+    let expect_system_memlock_usage = 16 * 1024; // TODO: depends on RING_SIZE
+                                                 // lower the softlimit such that the test will complete quickly but also don't
+                                                 // lower it so much that other tests will fail
+    let temp_softlimit: u64 =
+        2 * (max_number_of_systems_spawned_by_other_tests as u64) * expect_system_memlock_usage;
+    assert!(temp_softlimit <= hard);
+    nix::sys::resource::setrlimit(
+        nix::sys::resource::Resource::RLIMIT_MEMLOCK,
+        temp_softlimit,
+        hard,
+    )
+    .unwrap();
+    scopeguard::defer!({
+        nix::sys::resource::setrlimit(nix::sys::resource::Resource::RLIMIT_MEMLOCK, soft, hard)
+            .unwrap();
+    });
```

Comment:
Ok so the weirdness is "leaving space" for other tests with their systems. Makes sense, and test passed :) I'll check if it exited because locked memory is no longer needed with the runners, I already forgot their patch levels.

Reply:
Ah, of course we have no logging if the test passed, all good.